src/System.EventLog/src/System/Diagnostics/SharedUtils.cs

```
@@ -0,0 +1,335 @@
+                                            try
+                                            {
+                                                currentVersion[2] = GetLargestBuildNumberFromKey(k);
+
+                                                if (currentVersion[0] > largestVersion[0]
+                                                    || ((currentVersion[0] == largestVersion[0]) && (currentVersion[1] > largestVersion[1])))
+                                                {
+                                                    largestVersion = currentVersion;
+                                                }
+                                            }
+                                            finally
+                                            {
+                                                k.Close();
+                                            }
+                                        }
+                                    }
+
+                                    version = "v" + largestVersion[0] + "." + largestVersion[1] + "." + largestVersion[2];
+                                }
+                            }
+                            finally
+                            {
+                                policyKey.Close();
+                            }
+
+                            if (version != null && version != String.Empty)
+                            {
+                                StringBuilder installBuilder = new StringBuilder();
+                                installBuilder.Append(installRoot);
+                                if (!installRoot.EndsWith("\\", StringComparison.Ordinal))
+                                    installBuilder.Append("\\");
+                                installBuilder.Append(version);
+                                dllDir = installBuilder.ToString();
+                            }
+                        }
+                    }
+                }
+            }
+            catch
```

remove the empty catch block. it achieves nothing.